modules/rte_mis_allocation/src/Controller/StudentAdmissionReportController.php

```diff
@@ -0,0 +1,721 @@
+    else {
+      $currentUserRole = $this->currentUser->getRoles(TRUE);
+    }
+
+    $content = [];
+
+    if (array_intersect(['app_admin', 'state_admin'], $currentUserRole)) {
+      $content = $this->getStateAdminContent($id);
+    }
+    elseif (in_array('district_admin', $currentUserRole)) {
+      $content = $this->getDistrictAdminContent($id);
+    }
+    elseif (in_array('block_admin', $currentUserRole)) {
+      $content = $this->getBlockAdminContent($id);
+    }
+    else {
+      return [['#markup' => $this->t('No user account found!!')]];
+    }
+    return $content;
+  }
+
+  /**
+   * Get content for state admin.
+   */
+  protected function getStateAdminContent($id = NULL) {
+    // Implemented data fetching logic.
+    // Serial Number.
+    $serialNumber = 1;
+    $districts = $this->entityTypeManager()->getStorage('taxonomy_term')->loadTree('location', 0, 1, TRUE) ?? NULL;
+    $data = [];
+    if ($districts) {
+      foreach ($districts as $district) {
+        $blocks = $this->getBlocksCount($district->id());
+        $schools = count($this->getSchoolList($district->id()));
+        $total_rte_seats = $this->totalRteSeats('state_admin', $district->id());
+        $total_applications = $this->studentDetails('state_admin', $district->id());
+        $total_applied = $this->studentDetails('state_admin', $district->id(), 'applied');
+        $total_duplicate = $this->studentDetails('state_admin', $district->id(), 'duplicate');
+        $total_incomplete = $this->studentDetails('state_admin', $district->id(), 'incomplete');
+        $total_rejected = $this->studentDetails('state_admin', $district->id(), 'rejected');
+        $total_approved = $this->studentDetails('state_admin', $district->id(), 'approved');
+        $total_allotted = $this->studentStatus('state_admin', $district->id(), 'allotted');
+        $total_admitted = $this->studentStatus('state_admin', $district->id(), 'admitted');
+        $total_not_admitted = $this->studentStatus('state_admin', $district->id(), 'not_admitted');
+        $total_dropout = $this->studentStatus('state_admin', $district->id(), 'dropout');
+        $total_unallotted = $total_approved - ($total_allotted + $total_admitted + $total_not_admitted + $total_dropout);
+
+        // Create link render array.
+        $block_id = $district->id();
+        $url = Url::fromUri("internal:/allotment-reports/{$block_id}");
+        $link = Link::fromTextAndUrl($district->label(), $url)->toRenderable();
+
+        $data[] = [$serialNumber, ['data' => $link], $blocks, $schools, $total_rte_seats,
+          $total_applications, $total_applied, $total_duplicate, $total_incomplete,
+          $total_rejected, $total_approved, $total_allotted, $total_unallotted, $total_admitted, $total_not_admitted, $total_dropout,
+        ];
+        $serialNumber++;
+      }
+
+      return $data;
+    }
+    // Return a markup about missing location.
+    return [['#markup' => $this->t('No districts to display.')]];
```

Comment:
Can we return single array and not an array within array.

Reply:
ok